src/main/java/com/booleanuk/Poker.java

```diff
@@ -1,27 +1,122 @@
+        } else if (pairFirst) {
+            winningPair = firstHand;
+        } else if (pairSecond){
+            winningPair = secondHand;
+        }
+
+        return Arrays.equals(winningPair, new String[2])  ? new String[0] : winningPair;
     }
 
     // Extension 1
     public String[] winningPairFromArray(String[][] hands) {
         // Implement the winningPairFromArray logic here and return the array containing the winning pair to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+
+        String[] winningPair = {"0", "0"};
+        ArrayList<Integer> pairIndices = new ArrayList<>();
+
+        int i = 0;
+        for (String[] hand : hands){
+            if (Arrays.stream(hand).distinct().count() == 1) {
+                pairIndices.add(i++);
+            }
+        }
+
+        for (int pair : pairIndices){
+            if (valueMap.get(hands[pair][0]) > valueMap.get(winningPair[0])){
+                winningPair = hands[pair];
+            }
+        }
+
+        return Arrays.equals(winningPair, new String[]{"0", "0"}) ? new String[0] : winningPair;
     }
 
     // Extension 2
     public String[] winningThreeCardHand(String[][] hands) {
         // Implement the winningThreeCardHand logic here and return the array containing the winning hand to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+
+        String[] winningHand = {"0", "0", "0"};
+        ArrayList<Integer> pairs = new ArrayList<>();
+        ArrayList<Integer> threeOfAKinds = new ArrayList<>();
+
+        int i = 0;
+        for (String[] hand : hands){
+            long distinctCount = Arrays.stream(hand).distinct().count();
+
+            if (hand.length == 3 && distinctCount == 1) {
+                threeOfAKinds.add(i);
+            } else if (hand.length == 3 && distinctCount == 2) {
+                pairs.add(i);
+            } else if (hand.length == 2 && distinctCount == 1) {
+                pairs.add(i);
+            }
+            i++;
+        }
+        for (int threeOfAKind : threeOfAKinds) {
+            if (valueMap.get(hands[threeOfAKind][0]) > valueMap.get(winningHand[0])) {
+                winningHand = hands[threeOfAKind].clone();
+            }
+        }
+
+        if (Arrays.equals(winningHand, new String[]{"0", "0", "0"})) {
+            for (int pair : pairs) {
+                if (valueMap.get(hands[pair][0]) > valueMap.get(winningHand[0])) {
+                    winningHand = hands[pair].clone();
+                }
+            }
+        }
+        return Arrays.equals(winningHand, new String[]{"0", "0", "0"}) ? new String[0] : winningHand;
     }
 
+
     // Extension 3
     public String extensionThreeMethods() {
         // Completely replace this method with suitable methods to solve Extension 3
```

Comment:
Nice approach identifying pairs and handling card values using a map, and good effort on extension 2.